src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -21,25 +19,30 @@ public class IPAddress
 
         internal const long LoopbackMask = 0x00000000000000FF;
 
-        // IPv6 Changes: make this internal so other types that understand about
-        //               IPv4 and IPv4 can still access it rather than the obsolete property.
-        internal long Address;
-
-        internal string StringRepresentation;
-
         public static readonly IPAddress IPv6Any = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
         public static readonly IPAddress IPv6Loopback = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 }, 0);
         public static readonly IPAddress IPv6None = new IPAddress(new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }, 0);
 
-        /// <devdoc>
-        ///   <para>
-        ///     Default to IPv4 address
-        ///   </para>
-        /// </devdoc>
-        private readonly AddressFamily _family = AddressFamily.InterNetwork;
-        private readonly ushort[] _numbers = new ushort[NumberOfLabels];
-        private long _scopeId = 0;
-        private int _hashCode = 0;
+        /// <summary>
+        /// For IPv4 addresses, this field stores the Address.
+        /// For IPv6 addresses, this field stores the ScopeId.
```

Comment:
> I'm not currently convinced that the described alternatives would produce a net advantage in readability over the current approach.

And they'd have costs of their own, e.g. virtual calls where none exist today. I like the idea of a backing field accessed via properties with asserts.